src/components/FormSelectionContainer/FormSelect/FormSelect.tsx

```diff
@@ -14,38 +15,80 @@ interface FormSelectProps {
 export const FormSelect: FunctionComponent<FormSelectProps> = ({ id, form, onAddForm, ...props }) => {
   const { config } = useConfigContext();
   const [errorMsg, setErrorMsg] = useState<string | null>(null);
-  const [isValidDns, setIsValidDns] = useState<boolean>();
+  const [isValidDns, setIsValidDns] = useState<boolean>(true);
+  const [isValidOwner, setIsValidOwner] = useState<boolean>(true);
+  const [isValidEntry, setIsValidEntry] = useState<boolean>(true);
   const refButton = useRef<HTMLDivElement>(null);
 
   const handleForm = async (): Promise<void> => {
+    if (isValidEntry) {
+      onAddForm();
+    } else {
+      ReactTooltip.show(refButton.current as unknown as Element);
+    }
+  };
+
+  const dnsCheck = useCallback(async () => {
     if (config?.network === "local") {
       setIsValidDns(true); // for local e2e to pass, skip dns validate + set valid
     } else {
       const isDnsValidated = await validateDns(form);
-      if (!isDnsValidated) {
-        setErrorMsg(
-          `${getIssuerAddress(form.defaults)} address could not be found at ${getIssuerLocation(
-            form.defaults
-          )} DNS TXT records.`
-        );
-        ReactTooltip.show(refButton.current as unknown as Element);
-      }
       setIsValidDns(isDnsValidated);
     }
-  };
+  }, [config, form]);
 
-  useEffect(() => {
-    if (isValidDns) {
-      onAddForm();
+  const ownershipCheck = useCallback(async () => {
+    const wallet = config?.wallet;
+    const contractAddress = getIssuerAddress(form.defaults);
+    const isDID = checkDID(form.defaults);
+    if (isDID) {
+      setIsValidOwner(true);
+      return;
     }
-  }, [isValidDns, onAddForm]);
+    const contractType = form?.type;
+    if (config?.network === "local") {
+      setIsValidOwner(true); // for local e2e to pass, skip ownership validate + set valid
+    } else if (contractAddress != undefined && wallet != undefined) {
+      const valid = await checkOwnership(contractType, contractAddress, wallet);
```

Comment:
strict equality?

Reply:
yes, i've change all my equality to strict equality